src/clib/pio_darray.c

```diff
@@ -1961,8 +1985,44 @@ int PIOc_read_darray(int ncid, int varid, int ioid, PIO_Offset arraylen,
     pioassert(iodesc->rearranger == PIO_REARR_BOX || iodesc->rearranger == PIO_REARR_SUBSET,
               "unknown rearranger", __FILE__, __LINE__);
 
-    ios->io_fstats->rb += iodesc->mpitype_size * iodesc->llen;
-    file->io_fstats->rb += iodesc->mpitype_size * iodesc->llen;
+    /* Get var description. */
+    vdesc = &(file->varlist[varid]);
+
+    /* Run these on all tasks if async is not in use, but only on
+     * non-IO tasks if async is in use. */
+    if (!ios->async || !ios->ioproc)
+    {
+        spio_ltimer_stop(ios->io_fstats->rd_timer_name);
+        spio_ltimer_stop(ios->io_fstats->tot_timer_name);
+        spio_ltimer_stop(file->io_fstats->rd_timer_name);
+        spio_ltimer_stop(file->io_fstats->tot_timer_name);
+
+        /* Find out PIO data type of var. */
+        if ((ierr = PIOc_inq_vartype(ncid, varid, &vdesc->pio_type)))
```

Comment:
Same as above, if var type and type size (below) are cached we don't need this call.